stablehlo/dialect/VhloBytecode.cpp

```diff
@@ -696,6 +927,185 @@ void VhloBytecodeInterface::write(TypeExtensionsAttr attr,
+
+FloatV1Attr VhloBytecodeInterface::readFloatAttr(
+    DialectBytecodeReader &reader) const {
+  LOG_READ_CALL;
+  Type type;
+  if (failed(reader.readType(type))) return FloatV1Attr();
+  FailureOr<APFloat> value =
+      reader.readAPFloatWithKnownSemantics(getFloatSemantics(type));
+  if (failed(value)) return FloatV1Attr();
+
+  assertFromVhlo(type);
+  return FloatV1Attr::get(getContext(), type, *value);
+}
+
+void VhloBytecodeInterface::write(FloatV1Attr attr,
+                                  DialectBytecodeWriter &writer) const {
+  assertFromVhlo(attr.getType());
+  writer.writeVarInt(vhlo_encoding::kFloatAttr);
+  writer.writeType(attr.getType());
+  writer.writeAPFloatWithKnownSemantics(attr.getValue());
+}
+
+//===----------------------------------------------------------------------===//
+// FlatSymbolRefV1Attr
+
+FlatSymbolRefV1Attr VhloBytecodeInterface::readFlatSymbolRefAttr(
+    DialectBytecodeReader &reader) const {
+  LOG_READ_CALL;
+  Attribute rootReference;
+  if (failed(reader.readAttribute(rootReference))) return FlatSymbolRefV1Attr();
+
+  assertFromVhlo(rootReference);
+  return FlatSymbolRefV1Attr::get(getContext(), rootReference);
+}
+
+void VhloBytecodeInterface::write(FlatSymbolRefV1Attr attr,
+                                  DialectBytecodeWriter &writer) const {
+  assertFromVhlo(attr.getRootReference());
+  writer.writeVarInt(vhlo_encoding::kFlatSymbolRefAttr);
+  writer.writeAttribute(attr.getRootReference());
+}
+
+//===----------------------------------------------------------------------===//
+// IntegerV1Attr
+
+IntegerV1Attr VhloBytecodeInterface::readIntegerAttr(
+    DialectBytecodeReader &reader) const {
+  LOG_READ_CALL;
+  Type type;
+  if (failed(reader.readType(type))) return IntegerV1Attr();
+  assertFromVhlo(type);
+
+  // Extract the value storage width from the type.
+  unsigned bitWidth;
+  if (auto intType = type.dyn_cast<IntegerV1Type>()) {
+    bitWidth = intType.getValue().getWidth();
+    type = intType.getValue();
+  } else if (type.isa<IndexV1Type>()) {
+    bitWidth = IndexType::kInternalStorageBitWidth;
+    type = IndexType::get(getContext());
+  } else {
+    reader.emitError()
+        << "expected integer or index type for IntegerAttr, but got: " << type;
+    return IntegerV1Attr();
+  }
+
+  FailureOr<APInt> value = reader.readAPIntWithKnownWidth(bitWidth);
+  if (failed(value)) return IntegerV1Attr();
+  return IntegerV1Attr::get(getContext(), IntegerAttr::get(type, *value));
+}
+
+void VhloBytecodeInterface::write(IntegerV1Attr attr,
+                                  DialectBytecodeWriter &writer) const {
+  writer.writeVarInt(vhlo_encoding::kIntegerAttr);
+  LLVM_DEBUG(llvm::dbgs() << "IntegerAttr: " << attr << '\n');
+  auto intAttr = attr.getValue();
+  if (intAttr.getType().isa<IndexType>()) {
+    writer.writeType(IndexV1Type::get(getContext()));
+  } else {
+    assert(intAttr.getType().isa<IntegerType>());
+    // Wrap integer type in vhlo::IntegerV1Type
```

Comment:
Updated the comment. I think it is important because this is a non-standard piece, since this is one of the only times we wrap builtin types/attrs:

```
    // Note, since IntegerV1Type wraps builtin IntegerAttr, we need to explicitly
    // convert to IntegerV1Type here to avoid calling into builtin dialect bytecode
    // writer.
```